src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -571,6 +640,11 @@ protected override void Dispose(bool disposing)
                 public override int Read(Span<byte> destination)
                 {
                     Http2Stream http2Stream = _http2Stream ?? throw new ObjectDisposedException(nameof(Http2ReadStream));
+                    if (http2Stream._abortException != null)
+                    {
+                        throw http2Stream._abortException;
+                    }
```

Comment:
This will overwrite any call stack information and Watson information in the exception.  It'd be better to do `ExceptionDispatchInfo.Throw(http2Stream._abortException)`.

Reply:
Or, actually, shouldn't this be wrapped as the inner exception of an IOException?

Reply:
wrapping happens in SendAsync based on exception type. We may have _abortException because we got RST from server with specific code or because we detected invalid state machine. I think we will raise up IOException if we hit IO error.